codegen/cmd/injection-gen/generators/reconciler_controller.go

```diff
@@ -253,6 +253,7 @@ func NewImpl(ctx {{.contextContext|raw}}, r Interface{{if .hasClass}}, classValu
 
 	impl := {{.controllerNewImpl|raw}}(rec, logger, ctrTypeName)
 	agentName := defaultControllerAgentName
+	impl.Concurrency = controller.DefaultThreadsPerController
```

Comment:
We can drop this line no? Since we do the defaulting in here https://github.com/knative/pkg/pull/2160/files#diff-3c5bb5f48211873c58fcba055dcae2ac7b1958969219e06e1508d76d485dace7R252-R254

Reply:
@dprotaso Maybe we can't drop this, for controller author will use the `NewImpl` generated, and the `NewImpl` in `pkg/controller/controller.go` only used for test (as far as i know)

Reply:
I'm under the impression line 254

```
impl := {{.controllerNewImpl|raw}}(rec, logger, ctrTypeName)
```

This becomes `controller.NewImpl` which has the same defaulting logic

Reply:
yeah,see it , thanks, sorry for reply late